include/usb_cam/usb_cam.hpp

```diff
@@ -273,9 +276,11 @@ class UsbCam
   /// @brief Get pixel format from string. Required to have logic within UsbCam object
   /// in case pixel format class requires additional information for conversion function
   /// (e.g. number of pixels, width, height, etc.)
-  /// @param str name of supported format (see `usb_cam/supported_formats.hpp`)
+  /// @param pixFmtStr name of supported format (see `usb_cam/supported_formats.hpp`)
+  /// @param avDevFmtStr name of av device format (only used for mjpeg stream)
   /// @return pixel format structure corresponding to a given name
-  inline std::shared_ptr<pixel_format_base> set_pixel_format_from_string(const std::string & str)
+  inline std::shared_ptr<pixel_format_base> set_pixel_format_from_string(const std::string & pixFmtStr,
+                                                                         const std::string & avDevFmtStr)
```

Comment:
@boitumeloruf so I've tried really hard to keep the ffmpeg stuff separate from the rest of this base library. Right now the only place that has any ffmpeg stuff is in the mjpeg format file.

I dont know if I'd be OK with adding reference to it here.

Is there any other way we could get this av device format string to the conversion class?

Reply:
One idea would be to rename the function into just `set_pixel_format`and instead of passing the pixel format string to the function, one could pass the full parameter list in the form of the `parameter_t` struct to the function. In this way, one would have full access to all parameters, allowing one to pass the necessary parameters to the `pixel_format` class.

Reply:
@boitumeloruf good suggestion! Could you implement that here for this PR?